text/mavlink2_1.md

```diff
@@ -0,0 +1,62 @@
+  * Start date: 2022-10-12
+  * Contributors: Andrew Tridgell
+
+# Summary
+
+Rough proposal for MAVLink 2.1
+
+# Motivation
+
+Key features wanted:
+
+ * support for more than 256 system IDs
+ * support for checking CRCs in routers that don't have all xml
+ * better target_system/target_component support
+ * maximum compatibility with existing mavlink2 systems
+
+# Detailed Design
+
+The design takes advantage of the incompat_flags and compat_flags
+fields in the MAVLink2 header.
+
+New incompat_flags flags would be:
+
+ * MAVLINK_IFLAG_SYSID16 : system ID in header is 16 bits wide
+ * MAVLINK_IFLAG_TARGETTED : header has target_system and target_component appended
+ * MAVLINK_IFLAG_SPLIT_CRC : 16 bit crc is as described below
+
+New compat_flags flags would be:
+
+ * MAVLINK_FLAG_21_CAPABLE : sender is capable of handling MAVLink 2.1
+
+Note that for the SYSID16 and TARGETTED flags the sender should only
+use the flag if is it needed. So if the systemID in the message is
+<=255 then the flag should not be set and the existing 8 bit field
+should be used. For the TARGETTED bit this should only be used if the
+target systemID is >255 (needing 16 bits) or if the message that needs
+targetting doesn't have a target_system or target_component field.
+
+Following these rules maximises compatibility with existing MAVLink2
+capable systems.
+
+Note that when SYSID16 flag is set then the target_system would be 16
+bits when TARGETTED is set.
+
+For messages with existing target_system/target_component fields, if
+the TARGETTED bit is set then the existing fields are ignored by the
+recipient. The sender should set the old fields to zero.
+
+The 16 bit size of the expanded systemID could perhaps be 32 bits,
+allowing for an IPv4 address to be used, which could be useful for
+many use cases. The target audience is drone light shows which often
+use WiFi.
+
+## Split CRC
+
```

Comment:
Can we support omitting omit the CRC altogether? It isn't needed if running over a network that already does such checks - such as TCPIP.

Reply:
yes, that could be an incompat flag, but saying it "isn't needed" ignores the way we use it to ensure both ends of the link agree on the message structure

Reply:
I've been thinking of the router in an IP network - it doesn't need to check the CRC_EXTRA because it doesn't care what the message is, and it doesn't need to check the CRC because the IP network does that underneath.

But of course the message still needs the CRC_EXTRA at the endpoints. A router in this circumstance would benefit from being able to avoid running any CRC checks at all.

Reply:
I think we should *strongly* avoid dropping CRC's here. A corrupt partial packet can still be generated at the encoder, or network corruption (while rare) can actually happen. I'm not sure the bandwidth win is large enough to justify this.

Reply:
The CRCs are required in any case for the checks on the endpoint, so we can't remove them.

What would be good though is the ability to specify whether particular CRC checks are run. A router needs to be able to not test on the CRC_EXTRA. An endpoint should have the option to decide it does not want to run the full CRC if it things the link is fine. I think that the proposal addresses that.